lib/crypto.js

```diff
@@ -662,6 +672,32 @@ function filterDuplicates(names) {
   }).sort();
 }
 
+// This implements Brad Hill's Double HMAC pattern from
+// https://www.nccgroup.trust/us/about-us/
+//   newsroom-and-events/blog/2011/february/double-hmac-verification/.
+// In short, it's near-impossible to write a reliable constant-time compare in a
+// high level language like JS, because of the many layers that can optimize
+// away attempts at being constant time.
+//
+// Double HMAC avoids that problem by blinding the timing channel instead. After
+// running the inputs through a second round of HMAC, we are free to
+// short-circuit comparison, because the time it takes to reach the
+// short-circuit has no relation to the similarity between the guessed digest
+// and the correct one.
+exports.timingSafeEqual = timingSafeEqual;
+function timingSafeEqual(a, b) {
+  if (!(a instanceof Buffer && b instanceof Buffer)) {
+    throw new TypeError('Argument should be a Buffer');
```

Comment:
nit: Can you separate these checks so that the error messages can be made more specific / informative?

```js
if (!(a instanceof Buffer))
  throw new TypeError('First argument must be a Buffer');
if (!(b instanceof Buffer))
  throw new TypeError('Second argument must be a Buffer');
```
(note that I also changed the `should` into `must`)